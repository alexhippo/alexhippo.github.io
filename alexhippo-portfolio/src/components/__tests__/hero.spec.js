import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from 'gatsby'
import Hero from "../hero"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Alex Hipolito | Web Development, Test Automation, Agile Delivery`,
        }
      }
    })
  )
})

describe("Hero", () => {
  it("renders correctly", () => {
    const content = {
        "frontmatter": {
            "title": "I'm Alex.",
            "greetings": "Hello, kumusta ka?",
            "subtitlePrefix": "I ",
            "subtitleHighlight": "build, test and deliver."
          }
    };

    const tree = renderer
      .create(<Hero content={content} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})