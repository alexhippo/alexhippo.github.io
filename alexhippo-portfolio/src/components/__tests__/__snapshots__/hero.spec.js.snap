// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div
  className="jumbotron jumbotron-fluid"
  id="hero"
>
  <div
    className="container"
  >
    <section
      className="sc-bdfBwQ cyhnvE"
    >
      <h1
        className="title"
      >
        Hello, kumusta ka?
         
        I'm Alex.
      </h1>
      <h2
        className="subtitle"
      >
        I 
         
        <span
          className="highlighted"
        >
          build, test and deliver.
        </span>
        Â 
      </h2>
      <div
        className="description"
      />
    </section>
  </div>
</div>
`;
